mobile/src/components/common/search/search.tsx

```diff
@@ -0,0 +1,61 @@
+import { debounce } from 'debounce';
+import React, { FC } from 'react';
+import { TextInput } from 'react-native';
+
+import { AppColor } from '~/common/enums/enums';
+import { SearchInput } from '~/common/types/types';
+import { Icon, View } from '~/components/common/common';
+import { DEFAULT_SEARCH_TEXT } from '~/components/common/search/common/constants';
+import { useAppForm, useEffect, useFormControl, useState } from '~/hooks/hooks';
+
+import { styles } from './styles';
+
+type Props = {
+  onSearch: (text: string) => void;
+};
+
+const Search: FC<Props> = ({ onSearch }) => {
+  const [borderColor, setBorderColor] = useState('transparent');
+
+  const { watch, control } = useAppForm<SearchInput>({
+    defaultValues: DEFAULT_SEARCH_TEXT,
+  });
+
+  const { field } = useFormControl({ name: 'search', control: control });
+  const { value, onChange } = field;
+
+  const watchSearchText = watch('search');
+
+  const handleOnSearch = debounce(onSearch, 1000);
```

Comment:
Remove on in handlers names 
```suggestion
  const handleSearch = debounce(onSearch, 1000);
```